syntax = "proto3";
package packets;

// Define your packet messages. No empty messages allowed.
message ChatPacket { string msg = 1; }
message DenyPacket { string reason = 1; }
message DirectionPacket { float dx = 1; float dy = 2; }
message DisconnectPacket { string reason = 1; }
message LoginPacket { string username = 1; string password = 2; }
message OkPacket { string msg = 1; }
message PositionPacket { float x = 1; float y = 2; }
message RegisterPacket { string username = 1; string password = 2; }
// Add more packet messages here

// Define the main Packet message
message Packet {
    oneof type {
        ChatPacket chat = 1;
        DenyPacket deny = 2;
        DirectionPacket direction = 3;
        DisconnectPacket disconnect = 4;
        LoginPacket login = 5;
        OkPacket ok = 6;
        PositionPacket position = 7;
        RegisterPacket register = 8;
        // Add more packet types here
    }
}
